instrumentation/manticore/lib/opentelemetry/instrumentation/manticore/patches/response.rb

```diff
@@ -0,0 +1,103 @@
+          # @api #.call_with_otel_trace! - is the entry point to creating the OpenTelemetry
+          #   traces and spans if the class is loaded.
+          # it has direct to Manticore::Response methods and instance variables.
+          # Take great care of any unintentional mutations:
+          # [@request, @response, .on_complete] - are being invoked directly from Manticore::Response
+          def call_with_otel_trace!
+            wrapped_request = OpenTelemetry::Instrumentation::Manticore::Util::WrappedRequest.new(@request)
+            merged_attr = request_attr(wrapped_request)
+                            .merge(header_attr(wrapped_request.headers))
+                            .merge(OpenTelemetry::Common::HTTP::ClientContext.attributes)
+            span = tracer.start_span("HTTP #{wrapped_request.method}", attributes: merged_attr, kind: :client)
+            OpenTelemetry::Trace.with_span(span) do
+              OpenTelemetry.propagation.inject(wrapped_request)
+            end
+            on_complete do |response|
+              process_resp_attr(span, OpenTelemetry::Instrumentation::Manticore::Util::WrappedResponse.new(response))
+            end
+            call_without_otel_trace!
+          rescue ::Manticore::ManticoreException => e
+            span.set_attribute('http.response.status_code', 500)
+            span.set_attribute('http.response.status_text', 'Internal Server Error')
+            span.set_attribute('http.response.exception', e.message)
+            span.status = OpenTelemetry::Trace::Status.http_to_status(500)
+          ensure
+            span.finish
+          end
+
+          private
+
+          # @param [OpenTelemetry::Instrumentation::Manticore::Util::WrappedRequest] wrapped_request A request object that contains
+          #   information of an HTTP call to be made
+          # @return [Hash] returns request attributes to be added to span
+          def request_attr(wrapped_request)
+            uri = URI.parse(wrapped_request.uri)
+            attr = {
+              'library' => 'Manticore',
+              'component' => 'http',
+              'http.method' => wrapped_request.method,
+              'http.host' => uri.host
+            }
+            attr['http.path'] = uri.path unless uri.path.empty?
+            attr['http.query'] = uri.query if uri.query
+            attr
+          rescue StandardError => e
+            OpenTelemetry.logger.debug("Error while fetching request attributes: #{e}",)
+            {}
+          end
+
+          # @param [OpenTelemetry::Trace::Tracer.in_span] span A running span that has not been completed
+          # @param [OpenTelemetry::Instrumentation::Manticore::Util::WrappedResponse] response A wrapped response with limited exposed methods.
+          def process_resp_attr(span, response)
+            attr = {
+              'http.response.status_code' => response.status_code,
+              'http.response.status_text' => response.status_text
+            }
```

Comment:
These attributes should follow the HTTP semantic conventions: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md

Reply:
This is addressed. Thank you for the documentation links!